src/FormAlter/MediaFormAlter.php

```diff
@@ -0,0 +1,179 @@
+
+/**
+ * Alters the media form for mpx items to add "Reimport" button.
+ *
+ * @package Drupal\media_mpx\FormAlter
+ */
+class MediaFormAlter implements ContainerInjectionInterface {
+
+  use StringTranslationTrait;
+  use MessengerTrait;
+
+  /**
+   * The mpx media types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * Update Video Item service.
+   *
+   * @var \Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem
+   */
+  private $updateService;
+
+  /**
+   * The custom media mpx logger service.
+   *
+   * @var \Drupal\media_mpx\MpxLogger
+   */
+  private $logger;
+
+  /**
+   * MediaFormAlter constructor.
+   */
+  private function __construct(MpxMediaType $mpxMediaTypeRepository, UpdateVideoItem $updateService, MpxLogger $logger) {
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+    $this->updateService = $updateService;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('media_mpx.repository.mpx_media_types'),
+      $container->get('media_mpx.service.update_video_item'),
+      $container->get('media_mpx.exception_logger')
+    );
+  }
+
+  /**
+   * Implements hook_form_FORM_ID_alter().
+   */
+  public function alter(&$form, FormStateInterface $form_state, $form_id): void {
+    if ($this->alterAppliesToForm($form_state) === FALSE) {
+      return;
+    }
+
+    $form['actions']['reimport'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Reimport mpx data'),
+      '#submit' => [[$this, 'reimportCallback']],
+    ];
+  }
+
+  /**
+   * Callback for the 'reimport' button.
+   */
+  public function reimportCallback(array $form, FormStateInterface $formState) {
+    /* @var \Drupal\Core\Entity\ContentEntityForm $form_object */
+    $form_object = $formState->getFormObject();
+    $video = $form_object->getEntity();
+
+    if (!$id_field = $this->resolveIdFieldName($video)) {
+      $this->messenger()->addError($this->t('The mpx field @field is not mapped to any local field. The Media source must be configured
+          before using the Reimport feature', ['@field' => 'Media:id'])
+      );
+      return;
+    }
+
+    try {
+      if (($field = $video->get($id_field)) && !$field->isEmpty()) {
+        $mpx_id = $video->{$id_field}->value;
+        $this->updateVideoData((int) $mpx_id, $video->bundle());
+      }
+    }
+    catch (\InvalidArgumentException $e) {
+    }
+  }
+
+  /**
+   * Returns the name of the entity field holding the mpx ID.
+   *
+   * @param \Drupal\media\Entity\Media $video
+   *   The media entity for which to fetch the mpx ID field.
+   *
+   * @return string|null
+   *   The name of the field holding the mpx ID, or NULL if not configured.
+   */
+  private function resolveIdFieldName(Media $video):? string {
```

Comment:
I think instead call `\Drupal\media\MediaSourceInterface::getSourceFieldDefinition()` on the media entity.

Reply:
Not sure if I'm missing something as I'm slightly new to Media in D8, but I'm not able to do that. It still requires me to pass the MediaType, which I need to load for that.

Reply:
You can call `$media->bundle->entity` to get the media type - example at https://github.com/nbc-bravo/core/blob/fd294077e54a94275201474c3d7b3979c874b671/modules/media_library/src/Form/MediaLibraryUploadForm.php#L141-L143